File: apps/ios/myFuckingMusic/Models/Invitation.swift
import Foundation

/// Mirrors `Invitation` from packages/shared/src/types/invitation.ts
struct Invitation: Codable, Identifiable, Sendable {
    let id: Int
    let code: String
    let role: UserRole
    let scopeId: Int?
    let status: InvitationStatus
    let createdBy: Int
    let redeemedBy: Int?
    let expiresAt: Date
    let createdAt: Date
    let redeemedAt: Date?
}

enum InvitationStatus: String, Codable, Sendable {
    case pending = "PENDING"
    case redeemed = "REDEEMED"
    case expired = "EXPIRED"
    case revoked = "REVOKED"
}

File: packages/shared/src/types/invitation.ts
import { UserRole } from "../enums/roles.js";
import { InvitationStatus } from "../enums/roles.js";

export interface Invitation {
  id: number;
  code: string;
  role: UserRole;
  scopeId: number | null;
  status: InvitationStatus;
  createdBy: number;
  maxUses: number;
  usedCount: number;
  expiresAt: Date;
  createdAt: Date;
}

export interface InvitationCreate {
  role: UserRole;
  scopeId?: number;
  maxUses?: number;
  expiresAt?: Date;
}
